calculator/main.py

```diff
@@ -37,14 +51,23 @@ def add_new_operation(calc, operation):
     :param operation: Dict with the single operation to be added.
                       ie: {'add': add_function}
     """
-    pass
-
+    if not isinstance(operation, dict):
+        raise InvalidOperation()
+        
+    for key, value in operation.items():
+        calc['operations'][key] = value
+        
+    
 
 def get_operations(calc):
     """
     Returns the list of operation names supported by given calculator.
     """
-    pass
+    op_list = []
```

Comment:
Instead of iterating though and manually building a list here you could simply `return list(calc['operations'])`